site/content/docs/features/config-file.md

```diff
@@ -1 +1,632 @@
+
+_**ServiceAccessIAMRole**_  
+The IAM role that provides AWS credentials to the App Runner service so that it can pull the container image from [Amazon Elastic Container Registry (Amazon ECR)](https://docs.aws.amazon.com/ecr). You can create a new role (the default) or use an existing role. To use an existing role, set `CreateNew` to `false` and include the role's ARN in `RoleArn`.
+
+_**Cpu**_  
+The number of CPU units reserved for each instance of your App Runner service. The following values are valid:  
+-- "1024" (the default): 1 vCPU.  
+-- "2048": 2 vCPU.
+
+_**Memory**_  
+The amount of memory reserved for each instance of your App Runner service. Valid values are "2048" (the default), "3072", and "4096".
+
+_**EncryptionKmsKey**_  
+The ARN of the [AWS Key Management Service (KMS)](https://docs.aws.amazon.com/kms) key that's used to encrypt application logs.
+
+_**HealthCheckProtocol**_  
+The IP protocol that AWS App Runner uses to perform health checks for your service. Valid values are "HTTP" and "TCP" (the default).
+
+_**HealthCheckPath**_  
+The URL that health check requests are sent to. This parameter is valid only if `HealthCheckProtocol` is set to "HTTP".
+
+_**HealthCheckInterval**_  
+The time interval, in seconds, between health checks. The valid range of values if 1 through 20, and the default is 5.
+
+_**HealthCheckTimeout**_  
+The time, in seconds, to wait for the response before deciding that a health check failed. The valid range of values if 1 through 20, and the default is 2.
+
+_**HealthCheckHealthyThreshold**_  
+The number of consecutive checks that must succeed before AWS App Runner decides that the service is healthy. The valid range of values if 1 through 20, and the default is 3.
+
+_**HealthCheckUnhealthyThreshold**_  
+The number of consecutive checks that must fail before AWS App Runner decides that the service is unhealthy. The valid range of values if 1 through 20, and the default is 3.
+
+_**VPCConnector**_  
+The VPC connector that associates your App Runner service with a custom Amazon VPC. For more information about VPC connectors, see [Networking with App Runner](https://docs.aws.amazon.com/apprunner/latest/dg/network.html) in the [AWS App Runner Developer Guide](https://docs.aws.amazon.com/apprunner/latest/dg/what-is-apprunner.html).
+
+By default, the deployment tool will not create a new VPC connector, but expects the ID of an existing VPC connector in `VpcConnectorId`.  
+To create a new VPC connector, set `CreateNew` to `true` and provide the following information:
+
+* `VpcId`  
+The VPC ID that AWS App Runner should use when it associates your service with a custom Amazon VPC.
+* `Subnets`  
+A list of subnet IDs that AWS App Runner should use when it associates your service with a custom Amazon VPC. Specify IDs of subnets of a single Amazon VPC. AWS App Runner determines the Amazon VPC from the subnets you specify.

```

Comment:
The descriptions says: "AWS App Runner determines the Amazon VPC from the subnets you specify."
But what if the developer has also provided an ID in the `VpcId` parameter (above)?
Which does the tool favor or is this an error condition?

Reply:
The VpcId in app runner is only used to filter the Subnets and Security groups. It is more of a helper setting and not the setting that will directly be applied.